src/Control/RequestHandler.php

```diff
@@ -576,6 +576,17 @@ public function Link($action = null)
         return null;
     }
 
+    /**
+     * Get the absolute URL for this controller, including protocol and host.
+     *
+     * @param string $action See {@link Link()}
+     * @return string
+     */
+    public function AbsoluteLink($action = '')
```

Comment:
Can we guarantee the return type here?
```suggestion
    public function AbsoluteLink($action = ''): string
```

Reply:
Since this is going in 5.1 I intentionally didn't add any type hinting, as that would require subclasses (e.g. custom controllers) which have already implemented an `AbsoluteLink()` method to change their signature.

We don't normally worry about that when implementing new methods but given this is a method that we already know is commonly implemented I think it's worth taking into account.

Reply:
<slight sarcasm> We'll never get there to full types with this approach </> :-D but I get the reason here.